static/app/components/events/interfaces/crashContent/stackTrace/rawContent.tsx

```diff
@@ -170,15 +170,18 @@ export default function displayRawContent(
   exception?: ExceptionValue,
   hasSimilarityEmbeddingsFeature: boolean = false
 ) {
+  const rawFrames = data?.frames || [];
   const frames: string[] = [];
 
-  (data?.frames ?? []).forEach((frame, frameIdx) => {
-    if (
-      !hasSimilarityEmbeddingsFeature ||
-      (frame.inApp && hasSimilarityEmbeddingsFeature)
-    ) {
-      frames.push(getFrame(frame, frameIdx, platform));
-    }
+  const hasInAppFrames = rawFrames.some(frame => frame.inApp);
+  const shouldFilterOutSystemFrames = hasSimilarityEmbeddingsFeature && hasInAppFrames;
+
+  const framesToUse = shouldFilterOutSystemFrames
+    ? rawFrames.filter(frame => frame.inApp)
+    : rawFrames;
+
+  framesToUse.forEach((frame, frameIdx) => {
```

Comment:
nit: could replace this `forEach` and `push` with a `const frames = framesToUse.map(...)`

Reply:
Oh, good call. I was just riffing off of what was there, but you're right, that's cleaner.